keras/layers/embeddings.py

```diff
@@ -55,7 +55,12 @@ def __init__(self, input_dim, output_dim,
                  weights=None, dropout=0., **kwargs):
         self.input_dim = input_dim
         self.output_dim = output_dim
-        self.init = initializations.get(init)
+        #we can also init the word embedding use pre-trained embedding files
+        #in this way, the init parameters can be theano variable
+        if type(init) is str:
+            self.init = initializations.get(init)
```

Comment:
if you define your init as a function on global initializations.get will pick it up. No need to do this